core/src/main/java/org/apache/calcite/sql/SqlBasicCall.java

```diff
@@ -76,7 +77,8 @@ public SqlOperator getOperator() {
   }
 
   public List<SqlNode> getOperandList() {
-    return UnmodifiableArrayList.of(operands); // not immutable, but quick
+    // not immutable, but quick
+    return Collections.unmodifiableList(Arrays.asList(operands));
```

Comment:
Maybe worth adding the pattern to Util?
`public static <T> List<T> Util.unmodifiableList(T[] elems)`

Reply:
I wouldn't recommend it.   You then expect others to have to know that such a utility exists and it will end up with some mixture of both.

Reply:
`Util` is used all over the place and provides lots of common functions regarding Collections. 
If the concern is code consistency and preventing mixture, maybe we should look at how to prevent "bad pattern" using checkstyle and similar checker tools...

Reply:
Thanks for the feedback.  If that is what is required to get this change acceptable, then I will oblige.  Just let me know.

Reply:
I don't see extra use of `Collections.unmodifiableList()` to wrap arrays in code base, and the pattern was already to use `UnmodifiableArrayList`, so I don't think the codebase have a consistency issue. Also `Util` usage is fairly common in Calcite code base, and it is kind of expected for people to look at that class for any utility method.

As for unit test, just reuse the same as previously? I think it should work as-is